lib/LLVMIRCodeGen/LLVMBackend.cpp

```diff
@@ -528,22 +528,62 @@ bool LLVMBackend::isOpSupported(const NodeInfo &NI) const {
   }
 }
 
+/// Initialize the LLVM options using command line options.
+/// If target is not explicitly given we use the host attributes.
 LLVMBackendOptions::LLVMBackendOptions() {
-  // Initialize using command-line options by default.
-  arch_ = llvmArch;
-  target_ = llvmTarget;
-  cpu_ = llvmCPU;
+  if (llvmTarget.empty()) {
```

Comment:
@mciprian13 I don't think it is a good idea to use this logic here and set the values based on the host attributes, because there could be multiple backends derived from LLVMBackend and some of them may be used e.g. for cross-compilation and would like to use different attributes. But those backends would still use LLVMBackendOptions and this change would break them. 

Could this logic be moved e.g. into the constructor of e.g. CPUBackend or any other specific LLVMBackend-derived backend?

Note: Part of the problem is related to the fact that we use the same command-line options (target, arch, cpu, etc) by different backends derived from LLVMBackend. And those backends may have totally different rules for setting the defaults if some of these command-line options were not specified explicitly.

Reply:
@opti-mix I moved the logic of settting the defaults to CPUBackend constructor.